components/gitpod-db/go/cost_center.go

```diff
@@ -169,17 +169,17 @@ func (c *CostCenterManager) UpdateCostCenter(ctx context.Context, newCC CostCent
 		return CostCenter{}, status.Errorf(codes.InvalidArgument, "Unknown attribution entity %s", string(attributionID))
 	}
 
-	// Downgrading from stripe
-	if existingCC.BillingStrategy == CostCenter_Stripe && newCC.BillingStrategy == CostCenter_Other {
+	// Transitioning into free plan
+	if existingCC.BillingStrategy != CostCenter_Other && newCC.BillingStrategy == CostCenter_Other {
```

Comment:
Should this updated logic be covered by tests?

Reply:
The logic didn't change as there are only two different strategies. The point here is that it is not about not stripe but about going free (from anywhere). I'm working on a follow-up PR, where I introduce a third strategy for chargebee plans in cancellation mode. That PR comes with a test for this.